ddtrace/contrib/structlog/patch.py

```diff
@@ -59,7 +60,25 @@ def _w_get_logger(func, instance, args, kwargs):
     """
 
     dd_processor = [_tracer_injection]
-    structlog._config._CONFIG.default_processors = dd_processor + list(structlog._config._CONFIG.default_processors)
+    if _tracer_injection not in list(structlog._config._CONFIG.default_processors):
+        structlog._config._CONFIG.default_processors = (dd_processor +
+                                                        list(structlog._config._CONFIG.default_processors))
```

Comment:
Why do we inject the tracer injection processor during the call to `get_logger`, and not during the call to `patch`? This is essentially patching a global static variable in the `structlog` library so it might be more suitable there (and a bit more efficient, as we'll no longer need the `not in` test)

Reply:
Could you expand what you mean by patch? Or link the code where you think it's better to inject the processor?

Reply:
I meant in the `patch` function, right after [this line](https://github.com/DataDog/dd-trace-py/pull/8713/files#diff-abc40ca457456134ef2c0ba0398ab2298427752f53544dd13949b5f99eabb117R102)

Reply:
`patch` itself does not have the necessary information needed to overwrite `structlog` config. It just determines which part of the actual `structlog` library should have our wrapper function around it